xarray/coding/cftime_offsets.py

```diff
@@ -121,7 +122,35 @@ def __sub__(self, other):
         else:
             return NotImplemented
 
+    def _next_higher_resolution(self):
+        if type(self) is Day:
+            return Hour(self.n * 24)
+        if type(self) is Hour:
+            return Minute(self.n * 60)
+        if type(self) is Minute:
+            return Second(self.n * 60)
+        if type(self) is Second:
+            return Millisecond(self.n * 1000)
+        if type(self) is Millisecond:
+            return Microsecond(self.n * 1000)
+        # if type(self) is Microsecond:
+        #    return Nanosecond(self.n * 1000)
+        raise NotImplementedError(
+            "Could not convert to integer offset at any resolution"
+        )
+
     def __mul__(self, other):
+
+        if isinstance(other, float):
+            n = other * self.n
+            # If the new `n` is an integer, we can represent it using the
+            #  same BaseCFTimeOffset subclass as self, otherwise we need to move up
+            #  to a higher-resolution subclass
+            if np.isclose(n % 1, 0):
+                return type(self)(int(n))
+
+            new_self = self._next_higher_resolution()
```

Comment:
I first tried to implement `NotImplemented` here, but `__mul__` is recreated in subclasses for `MonthBegin` etc.

Reply:
thanks for the suggestion